src/range.cr

```diff
@@ -101,11 +104,16 @@ struct Range(B, E)
   # ```
   def each : Nil
     current = @begin
-    while current < @end
+    if current.nil?
+      raise ArgumentError.new("Can't each beginless range")
+    end
+
+    end_value = @end
+    while end_value.nil? || current < end_value
       yield current
       current = current.succ
     end
-    yield current if !@exclusive && current == @end
+    yield current if !@exclusive && !end_value.nil? && current == end_value
```

Comment:
`current == nil` will never be true, so i don't think this extra check is required

Reply:
Good thinking :-)

Reply:
Done!